wolang/builtInFunctions.go

```diff
@@ -4,27 +4,31 @@ import (
 	"fmt"
 )
 
-func plus(terms []interface{}) (result interface{}, err error) {
-	result = int(0)
+func plus(terms []DataType) (result Addable, err error) {
+	result = Float{0}
 
 	for _, v := range terms {
-		if val, ok := v.(int); !ok {
-			return nil, fmt.Errorf("error: '%v' is not a number I can add!", v)
+		if t, ok := v.(Addable); !ok {
+			return nil, fmt.Errorf("error: '%v' is not a number I can add!", v.GetValue())
 		} else {
-			result = result.(int) + val
+			if result, err = result.Add(t); err != nil {
```

Comment:
This will make `plus` always return a `Float`, even when adding  two `Integer`s. I'm okay with that if it's intended behavior, just want to make sure that it is :)

My preferred way to go about this would be to have result from numeric operations be `Integers` unless a `Float` type "poisons" an operation and then the result becomes float. But I don't want to impose that if you think it may not be suitable for a didactic approach.